expirationwatch/expiration_watcher.go

```diff
@@ -0,0 +1,162 @@
+package expirationwatch
+
+import (
+	"errors"
+	"reflect"
+	"sync"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ocdogan/rbt"
+	log "github.com/sirupsen/logrus"
+)
+
+// ExpiredItem represents an expired item returned from the Watcher
+type ExpiredItem struct {
+	ExpirationTimeSeconds int64
+	Hash                  common.Hash
```

Comment:
If we want this to be generic, `Hash` should just be `[]byte` or `string` and I would also suggest renaming it to `ID`.